### PR TITLE
use sbt-github-actions

### PR DESCRIPTION
**Pull request by amesgen**
Supersedes and closes #3825

See #3847 for more discussion.

### CODE REVIEW

**Comment by amesgen** [comment]
CI does not trigger *again*, I (still) have no idea why. The CI output for this commit is here: https://github.com/amesgen/http4s/actions/runs/363811493

File: .mergify.yml [hunk 1]
```diff
@@ -3,15 +3,15 @@ pull_request_rules:
...
-      - status-success=Scala 2.13.2, Java adopt@1.11
-      - status-success=Scalafix
-      - status-success=Project Site
-      - status-success=Doc Site
+      - status-success=Build and Test (ubuntu-latest, 2.13.3, adopt@1.8)
+      - status-success=Build and Test (ubuntu-latest, 2.13.3, adopt@1.11)
+      - status-success=Build and Test (ubuntu-latest, 2.13.3, adopt@1.15)
+      - status-success=Build and Test (ubuntu-latest, 2.12.12, adopt@1.8)
+      - status-success=Build and Test (ubuntu-latest, 2.12.12, adopt@1.11)
+      - status-success=Build and Test (ubuntu-latest, 2.12.12, adopt@1.15)
```
Comment (rossabaker) [inline .mergify.yml:12]: We haven't historically blocked mergify on 1.15, which we're testing just to get out ahead of any problems. But I don't mind adding it as a gate.

**Review by rossabaker — APPROVED** [review]
Very nice work!

File: .mergify.yml (hunk 1 above)
Comment (amesgen) [reply .mergify.yml:12]: Ah, didn't know that. If any annoying problems occur on Java 15, we can remove disable it again. :+1:

**Comment by rossabaker** [comment]
I'm curious whether subsequent PRs get tested after we merge this.

**Comment by rossabaker** [comment]
I expect we'll have a couple annoying issues coming out of this, but it's definitely the way forward, and the status quo falls apart tomorrow.

**Pull request merged**